File: communication/mosquitto/simple-demo/go-server/config/mqtt.go
package config

import (
	"log"

	mqtt "github.com/eclipse/paho.mqtt.golang"
)

type MQTT struct {
	Client mqtt.Client
}

func NewMQTT() *MQTT {
	opts := mqtt.NewClientOptions()
	opts.AddBroker("tcp://localhost:1883")
	opts.SetClientID("go-server")

	client := mqtt.NewClient(opts)
	if token := client.Connect(); token.Wait() && token.Error() != nil {
		log.Fatal(token.Error())
	}

	return &MQTT{
		Client: client,
	}
}

func (m *MQTT) Close() {
	m.Client.Disconnect(250)
}

File: communication/mosquitto/simple-demo/python-server/mosquitto_python_server/config/mqtt.py
import paho.mqtt.client as mqtt


class MQTT:
    def __init__(self):
        self.client = mqtt.Client()
        self.client.connect("localhost", 1883)
        self.client.loop_start()

    def close(self):
        self.client.loop_stop()
        self.client.disconnect()
